app/controllers/articles_controller.rb

```diff
@@ -35,13 +40,25 @@ def update
   end
 
   def destroy
-    @article = Article.find(params[:id])
     @article.destroy
     redirect_to articles_path
   end
 
   private
 
+    def set_article
+      @article = Article.find(params[:id])
+    end
+
+    def correct_user
+      @article = current_user.articles.find_by(id: params[:id])
```

Comment:
not needed at all, try to use existing `@article` (it was already initialised on `set_article`)